Automattic/wp-calypso
# Update domain SSL card

## Proposed Changes
This PR introduces some updates in the domain settings SSL card.
If there are some issues that prevent the certificate from being issued, the card shows some specific messages to the user. It also introduces a button to force SSL certificate provisioning.

Depends on D157659-code

Here are new card configurations, depending on SSL certificate status and related issues.

### New domain registrations
The certificate status is pending an the button to request a new certificate is hidden (since DNS are not resolving the domain yet)
<img width="725" alt="Screenshot 2024-08-14 alle 09 44 01" src="https://github.com/user-attachments/assets/18256dc2-c4ca-4016-8b3a-8cf93ea3f4d8">

### Active certificate
<img width="742" alt="Screenshot 2024-08-07 alle 19 10 18" src="https://github.com/user-attachments/assets/5f7fc868-65b5-465f-b77d-0e16761cb55b">

### Certificate issues
We show a list of issues and a button to request a new certificate
<img width="746" alt="Screenshot 2024-08-07 alle 18 54 45" src="https://github.com/user-attachments/assets/55121e36-a28f-4ed2-aff2-118e6e8731d6">

### Expired domain
<img width="732" alt="Screenshot 2024-08-07 alle 18 58 25" src="https://github.com/user-attachments/assets/f536e3e6-d077-4b27-a3e8-f12af42a8a33">


## Why are these changes being made?
We want to improve te user experience, giving users a way to easily fix common SSL certificate issues.

## Testing Instructions
The best way to test the PR is to connect a new external domain with misconfigured dns (i.e. CAA records), verify that the UI show the correct message and request a new certificate before and after fixing dns. In the first case you will see a failed request notification, in the second one the certificate should be issued and the UI should show active status.

You can also hack the backend and send different statuses and messages. 
 
## Pre-merge Checklist
